code/modules/reagents/chemistry/reagents.dm

```diff
@@ -61,6 +62,7 @@
 
 // Called when this reagent is first added to a mob
 /datum/reagent/proc/on_mob_add(mob/M)
+	mobeffectactive = TRUE
```

Comment:
You need to make sure that everything that overrides on_mob_add also calls ..()

Reply:
they all do

Reply:
and they just need to not hard return all procs call their parent at the end if they havent 1. alredy manualy called it 2. returned a value 3. hit the end of proc/whatever() and auto-returned null

Reply:
>all procs call their parent at the end if they havent

This is false